File: go/fiber/internal/modules/address/module.go
package address

import (
	"database/sql"
	auth "tcc/internal/modules/auth/middleware"
	"tcc/internal/modules/client"

	"github.com/gofiber/fiber/v2"
)

func AddressModule(router *fiber.App, db *sql.DB) {
	clientRepository := client.NewClientRepository(db)
	addressRepository := AddressRepository{
		db: db,
	}
	service := AddressService{
		repository: struct{client client.ClientRepository; address AddressRepository}{
			client: clientRepository,
			address: addressRepository,
		},
	}
	controller := AddressController{
		service: service,
	}

	group := router.Group("/enderecos")
	{
		group.Post("", auth.AuthMiddleware(), controller.Create)
	}
}

File: rust/axum/src/modules/address/module.rs
use axum::{Router, routing::post, middleware::from_fn};
use super::controller::AddressController;
use crate::modules::auth::middleware::auth_middleware;

pub fn address_module() -> Router {
    let router = Router::new()
        .route("/", post(AddressController::create))
        .layer(from_fn(auth_middleware));

    Router::new().nest("/enderecos", router)
}